### PR TITLE
Made compatibility with `lark` ≥1.0!

### PR DESCRIPTION
**Pull request by KOLANICH**

### CODE REVIEW

File: hcl2/parser.py [hunk 1]
```diff
@@ -36,11 +36,10 @@ def create_parser_file():
     """
     lark_file = os.path.join(dirname(__file__), 'hcl2.lark')
     with open(lark_file, 'r') as lark_file, open(PARSER_FILE, 'w') as parser_file:
-        lark_inst = Lark(lark_file.read(), parser="lalr", lexer="standard")
+        lark_inst = Lark(lark_file.read(), parser="lalr", lexer="contextual")
 
-        data, memo = lark_inst.memo_serialize([TerminalDef, Rule])
-
-        print(PARSER_FILE_TEMPLATE % (data, memo), file=parser_file)
+        import lark.tools.standalone
```
Comment (rddimon) [inline hcl2/parser.py:41]: This import should be moved to the top

**Comment by KOLANICH** [comment]
>This import should be moved to the top

No, it shouldn't. End users are not supposed to call it.

**Comment by KOLANICH** [comment]
That branch should be taken only on the machines of the ones building a wheel from source. If it is taken in an installed wheel, it is likely it will error - `/usr/lib/python3.10/dist-packages` (the path on Debian-derived distros when packages are system-installed) is not available for writing.

**Comment by rddimon** [comment]
Hi @KOLANICH

Will you be able to rebase with the master and fix PR checks?

**Comment by KOLANICH** [comment]
>Will you be able to rebase with the master

already done.

>fix PR checks?

Works for me locally (`pytest ./test`). IDK why doesn't it work on the server, the CI pipeline seems to be non reporting what exactly is wrong. Feel free to fix the CI yourself.

**Comment by rddimon** [comment]
https://github.com/amplify-education/python-hcl2/actions/runs/3794008621/jobs/6451675562

**Comment by rddimon** [comment]
@htorianik will help with it

**Comment by KOLANICH** [comment]
Under "not reporting" I meant this: https://github.com/amplify-education/python-hcl2/actions/runs/3794168036/jobs/6451977387

```
py310-unit: commands[1]> nose2 --config tox.ini --verbose
2022-12-28T13:05:12.7068749Z test_load_terraform (test.unit.test_load.TestLoad)
2022-12-28T13:05:16.2686751Z Test parsing a set of hcl2 files and force recreating the parser file ... ok
2022-12-28T13:05:16.2692288Z test_load_terraform_from_cache (test.unit.test_load.TestLoad)
2022-12-28T13:05:16.3123322Z Test parsing a set of hcl2 files from a cached parser file ... ok
2022-12-28T13:05:16.3129150Z
2022-12-28T13:05:20.7365920Z ----------------------------------------------------------------------
2022-12-28T13:05:20.7366718Z Ran 2 tests in 3.606s
2022-12-28T13:05:20.7458974Z
2022-12-28T13:05:20.7459146Z FAILED
2022-12-28T13:05:20.7459476Z Name Stmts Miss Branch BrPart Cover Missing
2022-12-28T13:05:20.7460100Z -----------------------------------------------------------------
2022-12-28T13:05:20.7460545Z hcl2/__init__.py 5 2 0 0 60% 5-6
2022-12-28T13:05:20.7460829Z hcl2/api.py 6 0 0 0 100%
2022-12-28T13:05:20.7468841Z hcl2/lark_parser.py 1932 942 641 45 45% 49, 75-86, 94-137, 140-143, 151-161, 165-167, 178-197, 201-207, 218-234, 239-241, 244-249, 260-265, 294->297, 321-322, 325-334, 350-351, 366-367, 370, 374, 384, 389-390, 398-429, 447, 461-463, 467-469, 472, 475, 478-487, 491, 495, 498-510, 513-516, 519, 522, 526-537, 541-550, 554, 558, 576, 588-606, 609, 618, 623-638, 641-651, 654-663, 666-667, 671, 678, 682, 686, 691-706, 714-720, 728, 731-733, 739, 746, 749-752, 761-800, 806-807, 815, 819, 822, 830-832, 836-838, 846-851, 855-861, 874, 877-882, 885, 889, 892, 899-903, 909-914, 922-929, 932, 935-944, 947-952, 956, 958, 960, 962, 967-988, 1002, 1005-1006, 1009, 1015, 1020, 1029-1030, 1034, 1037, 1056-1060, 1063, 1079-1084, 1090, 1093, 1101, 1116-1118, 1121, 1126, 1129, 1133, 1138, 1143, 1147, 1161, 1179, 1184, 1200-1203, 1206, 1209-1213, 1245, 1248->exit, 1262-1266, 1287, 1291, 1294-1301, 1304, 1315, 1318, 1321, 1325, 1343-1346, 1373-1375, 1378-1379, 1396, 1400->1394, 1433, 1436-1438, 1467-1470, 1473, 1491, 1502, 1506, 1527-1537, 1558-1563, 1577-1579, 1586-1589, 1604, 1609-1610, 1655-1666, 1690-1700, 1706, 1722-1727, 1742, 1748-1749, 1752-1790, 1793-1802, 1806, 1808, 1812, 1817-1819, 1822-1835, 1842-1858, 1889-1892, 1908, 1918-1920, 1923-1946, 1950-1953, 1960-1961, 1964-1993, 1998-2004, 2008-2014, 2052, 2061, 2063, 2071, 2081-2087, 2097, 2100, 2105-2134, 2167-2169, 2172, 2181, 2194-2196, 2239, 2252-2267, 2274, 2276, 2289-2297, 2319-2331, 2336-2345, 2373-2377, 2384-2386, 2393-2395, 2400, 2406, 2408-2409, 2423-2427, 2432->exit, 2447, 2452-2453, 2456-2457, 2462-2463, 2472-2474, 2484, 2584, 2633, 2638, 2646, 2650, 2655-2656, 2659-2660, 2663, 2693-2898, 2901, 2906-2911, 2917->2926, 2929-2932, 2942-2945, 2950-2951, 2967, 2975, 3004-3008, 3013-3018, 3021, 3027-3035, 3039, 3043, 3060-3062, 3065-3083, 3086-3102, 3105-3107, 3113, 3118, 3123, 3128, 3133, 3138, 3143
2022-12-28T13:05:20.7470836Z hcl2/parser.py 18 1 4 1 91% 46
2022-12-28T13:05:20.7471589Z hcl2/transformer.py 166 10 74 10 91% 18, 32, 36, 91->93, 108, 121, 126, 167, 176, 193, 203, 260->263 2022-12-28T13:05:20.7472658Z hcl2/version.py 2 0 0 0 100%
2022-12-28T13:05:20.7473326Z -----------------------------------------------------------------
2022-12-28T13:05:20.7563524Z TOTAL 2129 955 719 56 49%
2022-12-28T13:05:20.7563738Z
2022-12-28T13:05:20.7801060Z py310-unit: exit 1 (8.61 seconds) /home/runner/work/python-hcl2/python-hcl2> nose2 --config tox.ini --verbose pid=2012
2022-12-28T13:05:20.7840611Z py310-unit: FAIL ✖ in 22.56 seconds
2022-12-28T13:05:20.7841351Z py311-unit: skipped because could not find python interpreter with spec(s): py311
2022-12-28T13:05:20.7874643Z lint: SKIP (0.74 seconds)
2022-12-28T13:05:20.7875119Z py37-unit: SKIP (0.00 seconds)
2022-12-28T13:05:20.7875429Z py38-unit: SKIP (0.00 seconds)
2022-12-28T13:05:20.7875758Z py39-unit: SKIP (0.00 seconds)
2022-12-28T13:05:20.7876198Z py310-unit: FAIL code 1 (22.56=setup[0.17]+cmd[13.78,8.61] seconds)
2022-12-28T13:05:20.7876568Z py311-unit: SKIP (0.00 seconds)
2022-12-28T13:05:20.7876876Z evaluation failed :( (23.56 seconds)
2022-12-28T13:05:20.8339598Z ##[error]Process completed with exit code 255.
```

No exception, no failed test name, no details, just something somewhere has failed.

**Comment by htorianik** [comment]
@KOLANICH nose2 requires coverage >80% like set in .coveragerc: https://github.com/amplify-education/python-hcl2/blob/f4872c3bde6ec0cebb8c09d5c34e19b019c45fbc/.coveragerc#L7

Give me a second, I'll fix it. It happened because lark_parser.py wasn't ignored for some reason.

File: .github/workflows/pr_check.yml [hunk 2]
```diff
@@ -20,8 +20,12 @@ jobs:
...
-          run: python -m pip install --upgrade "tox>=4.0.9,<5" build
+          run: python -m pip install --upgrade "tox>=4.0.9,<5" build lark
+        - name: Generate parser file as a side effect
+          run: python -m hcl2.parser
         - name: Build to generate the version file
           run: python3 -m build
+        - name: Install package
+          run: python3 -m pip install --upgrade ./dist/*.whl
         - name: Test with Tox
           run: tox
```
Comment (rddimon) [inline .github/workflows/pr_check.yml:31]: @KOLANICH For what are those changes needed?

**Comment by KOLANICH** [comment]
Done. 🎉🎉🎉 I guess you may want to report the bugs in `mypy` and `lark`.

File: .github/workflows/pr_check.yml [hunk 3]
```diff
@@ -20,7 +20,9 @@ jobs:
           with:
             python-version: ${{ matrix.python-version }}
         - name: Install dependencies
-          run: python -m pip install --upgrade "tox>=4.0.9,<5" build
+          run: python -m pip install --upgrade "tox>=4.0.9,<5" build lark
+        - name: Generate parser file as a side effect
+          run: python -m hcl2.parser
```
Comment (rddimon) [inline .github/workflows/pr_check.yml:25]: Do we need this change?
I believe, during the tox run this file will be generated automatically
Could you please try to remove it?

File: pyproject.toml [hunk 4]
```diff
@@ -22,7 +22,7 @@ classifiers = [
     "Programming Language :: Python :: 3.10",
 ]
 requires-python = ">=3.7.0"
-dependencies = ["lark>=0.10.0,<0.11.0"]
+dependencies = ["lark>=0.10.0"]
```
Comment (rddimon) [inline pyproject.toml:25]:
```suggestion
dependencies = ["lark>=0.10.0,<2"]
```

Comment (rddimon) [reply pyproject.toml]: I believe even better to set it to `dependencies = ["lark>=1,<2"]` 
We will release a new version with this new requirement

File: .github/workflows/pr_check.yml [hunk 5]
```diff
@@ -20,7 +20,7 @@ jobs:
           with:
             python-version: ${{ matrix.python-version }}
         - name: Install dependencies
-          run: python -m pip install --upgrade "tox>=4.0.9,<5" build
+          run: python -m pip install --upgrade "tox>=4.0.9,<5" build lark
```
Comment (rddimon) [inline .github/workflows/pr_check.yml:23]: Do we need `lark` package here?
During the tox run, we do `pip install -r requirements.txt` that will handle this install

**Comment by KOLANICH** [comment]
2BH I'm not sure how to behave about that and if the parsers generated by `lark==1.15` are compatible with `lark==1.0`. It needs some testing.

**Comment by rddimon** [comment]
> 2BH I'm not sure how to behave about that and if the parsers generated by `lark==1.15` are compatible with `lark==1.0`. It needs some testing.

I just wanted to highlight it

**Comment by KOLANICH** [comment]
Tested. `0.` versions are incompatible with the generated parser, `1.` are compatible.

**Review by htorianik — APPROVED** [review]

**Review by rddimon — APPROVED** [review]

**Comment by rddimon** [comment]
@KOLANICH Nice work!

Thank you for your contribution!

**Comment by KOLANICH** [comment]
In theory we can support both `1.` and `0.` with a little dispatching based on version.

**Comment by rddimon** [comment]
In my opinion, it's not needed
People can use `python-hcl2<4.2.0` or update it in another case

**Pull request merged**